### ISSUE TITLE
Missing support for Gemfile source declaration with a block

### ISSUE THREAD

**Issue by jimherz** [thoughtbot/appraisal#99]
For gems whose Gemfile has multiple source declarations, bundler (i.e., v1.10.4) emits the following security warning:

```
Warning: this Gemfile contains multiple primary sources. Using `source` more than once without a block is a security risk, and may result in installing unexpected gems. To resolve this warning, use a block to indicate which gems should come from the secondary source. To upgrade this warning to an error, run `bundle config disable_multisource true`.
```

However, source declarations with blocks are not supported by the appraisal gem when it generates appraisal gemfiles.  Gems declared inside of a source block do not appear in generated appraisal gemfiles.

**Comment by sikachu** [comment]
Thanks for reporting. I think we want to take care of that. 




Can you give me an example Appraisals file and Gemfile so I can use it as a failing test case, and the expected outcome on the generated gemfile? Thanks!

**Comment by jimherz** [comment]
Thanks.  I created a test gem: https://github.com/jimherz/test_source_block

I manually modified the generated gemfiles for the 2 appraisals so they reflect the correct source block usage.

Let me know if you need anything else.

**Comment by sikachu** [comment]
Ah, I see what you mean now. So source block should be preserved just like how we handle `path`, `git`, etc.

I'll work on adding a support for this.

**Comment by sikachu** [comment]
Sorry for kept you waiting. I just implemented this and release v2.1.0.
